futures-util/src/stream/futures_unordered/iter_pin_mut.rs

```diff
@@ -0,0 +1,36 @@
+use std::marker::PhantomData;
+use std::mem::PinMut;
+
+use super::FuturesUnordered;
+use super::node::Node;
+
+#[derive(Debug)]
+/// Mutable iterator over all futures in the unordered set.
+pub struct IterPinMut<'a, F: 'a> {
```

Comment:
Rather than duplicating the logic between `IterPinMut` and `IterMut`, I'd implement `IterMut` in terms of `PinMut`. The `Unpin` bound will make it safe to go from `PinMut` of the yielded value to `&mut` of the value.

Reply:
Done